wdl-format/src/format/comments.rs

```diff
@@ -0,0 +1,153 @@
+//! Format comments in a WDL file.
+//!
+//! All comments will be treated as either "preceding" or "inline" comments.
+//! Every comment will "belong" or "be owned" by a specific element in the
+//! syntax tree. If that element is moved from one place to another, the
+//! comment will move with it. Only syntax elemnts that are either the first
+//! element of a line or the last element of a line can own comments. Elemnts
+//! may span multiple lines, only the beginning (in the case of preceding
+//! comments) or the end (in the case of inline comments) of the element
+//! are considered.
+//!
+//! A preceding comment is a comment that appears on a line before an element.
+//! There may be any number of preceding comments and they may be separated
```

Comment:
Hmm... I'm not sure that comments separated by blank lines should be considered a single unit.

Reply:
For example something like this: https://github.com/stjude/seaseq/blob/3b6b503292a4ebb2c8e9053d10e04ffad82b3bad/seaseq-case.wdl#L400-L414. The `section` header isn't indented, but if it were, I'm not sure we should delete the blank lines and attach everything to `File sample_bam`.

Reply:
That's a great example. You're right that the current behavior is wrong for that case. But I'm not sure what the right approach is. I need to think on this case for a bit.

Reply:
I think the best approach I can think of for this case would be to leave blank lines as they appear. I don't see a way around "attaching" comments to non-trivia. It's required unless we want to visit _each and every_ comment to ensure it's being included somewhere. So I think no-matter-what, all of those `seaseq` comments should be "attached" to `File sample_bam`. But we can respect the blank lines. How does that sound?

Reply:
That seems like a good compromise. No issue with attaching them to the element, as long as we leave blank lines.